src/science/casa-cnp/casa_cnp.F90

```diff
@@ -1027,14 +1026,15 @@ SUBROUTINE casa_delplant(veg,casabiome,casapool,casaflux,casamet,            &
                                       * casabiome%ftransNPtoL(veg%iveg(npt),leaf)
       ENDIF
 
-      casapool%dNplantdt(npt,wood) = 0.0
-#ifndef ESM15
-      ! offline/trunk uses this condition
-      IF (casamet%lnonwood(npt)==0)                                            & 
-#endif
-      casapool%dNplantdt(npt,wood) = - casaflux%kplant(npt,wood)             &
+      !R. Law 25/10/24 removed ESM15 case as no need to exclude the condition
+      !that is in offline/trunk. Also re-write as IF / THEN
```

Comment:
Just a general comment: I would write this in the commit message rather than the actual source code as a commit message should explain why you made the change. I try to reserve code comments for explaining why the code does what it does to help future developers looking at the code (e.g. why we check for `casamet%lnonwood(npt)==0` when setting `casapool%dNplantdt(npt,wood)`).

Reply:
Thanks @SeanBryan51 for the feedback - I will keep this in mind for future code comments/commit messages. One of the challenges in this case was that there was no history as to when and why the condition had been added.